qutip/solver/options.py

```diff
@@ -257,8 +257,8 @@ class McOptions:
         "mc_corr_eps": 1e-10,
 
         "map_options": {
-            'num_cpus': multiprocessing.cpu_count(),
-            'timeout': threading.TIMEOUT_MAX,
-            'job_timeout': threading.TIMEOUT_MAX,
+            'num_cpus': None,
+            'timeout': None,
+            'job_timeout': None,
         },
```

Comment:
Would it work to have something on the lines:
```
options = {...}
options["map_options"] = qutip.solver.parallel.default_map_kw
```
?
This way both are the same and you do not need `_read_map_kw`.

Reply:
While it could work, I like the added sturdiness of `_read_map_kw`.
It allow to have the default to `None` instead of a mutable `dict` and serve a check that all desired key are there.
So is someone did `options["map_options"] ={'num_cpus': 2}`, it would still work.
Having `options["map_options"]` itself being a dict is risky since once overwritten, you can forget which keys it's supposed to have...